scipy/sparse/linalg/_isolve/_gcrotmk.py

```
@@ -283,11 +283,10 @@ def gcrotmk(A, b, x0=None, tol=1e-5, maxiter=1000, M=None, callback=None,
         raise ValueError("Invalid value for 'truncate': %r" % (truncate,))
 
     if atol is None:
-        warnings.warn("scipy.sparse.linalg.gcrotmk called without specifying `atol`. "
-                      "The default value will change in the future. To preserve "
-                      "current behavior, set ``atol=tol``.",
-                      category=DeprecationWarning, stacklevel=2)
         atol = tol
```

This should now raise an error, not set `atol = tol`.